msteams-platform/tabs/how-to/using-fluid-msteam.md

```diff
@@ -0,0 +1,317 @@
+
+```ts
+const [fluidContainer, setFluidContainer] = useState<IFluidContainer | undefined>(undefined);
+```
+
+1. Replace `TODO 4` with the following code. Here you'll parse the URL to get the query parameter string, defined by `containerIdQueryParamKey`, and retrieve the container ID. With the container ID, you can now load the container to get the container. Once you have the container, set the `fluidContainer` React state, defined above.
+
+```ts
+const getFluidContainer = async (url : URLSearchParams) => {
+    const containerId = url.get(containerIdQueryParamKey);
+    if (!containerId) {
+        throw Error("containerId not found in the URL");
+    }
+    const container = await getContainer(containerId);
+    setFluidContainer(container);
+};
+```
+
+1. Replace `TODO 5` with the following code. Now that you've defined how to get the Fluid container, you need to tell React to call `getFluidContainer` on load, and then store the result in state based on if the application is inside Teams.
+React's [useState hook](https://reactjs.org/docs/hooks-state.html) will provide the storage needed, and [useEffect](https://reactjs.org/docs/hooks-effect.html) will allow you to call `getFluidContainer` on render, passing the returned value into `setFluidContainer`.
+
+By adding `inTeams` in the dependency array at the end of the `useEffect`, the app ensures that this function only gets called on content page load.
+
+```ts
+useEffect(() => {
+    if (inTeams === true) {
+        microsoftTeams.settings.getSettings(async (instanceSettings) => {
+            const url = new URL(instanceSettings.contentUrl);
+            getFluidContainer(url.searchParams);
+        });
+        microsoftTeams.appInitialization.notifySuccess();
+    }
+}, [inTeams]);
+```
+
+1. Replace `TODO 6` with the following code. Note here that it's important to ensure that the content page is loaded inside Microsoft Teams and that the Fluid container is defined before passing it into the React component (defined as `FluidComponent` below).
+
+```ts
+if (inTeams === false) {
+  return (
+      <div>This application only works in the context of Microsoft Teams</div>
+  );
+}
+
+if (fluidContainer !== undefined) {
+  return (
+      <FluidComponent fluidContainer={fluidContainer} />
+  );
+}
+
+return (
+  <div>Loading FluidComponent...</div>
+);
+```
+
+### Create React component for Fluid view and data
+
+Now that you've married the basic creation flow of Teams and Fluid, you can now create your own React component that handles the interactions between the application view and Fluid data. From this point on, the logic and flow behaves just like any other Fluid-powered application. With the basic structure established, you can create any of the [Fluid examples](https://github.com/microsoft/FluidExamples) as a Teams application by changing the `ContainerSchema` and the application view's interaction with the DDSes/data objects on the content page.
```

Comment:
The term "married" sounds kind of funny in technical documentation :-P 

How about: Now that you've **integrated** the basic creation flow of Teams and Fluid, ....